modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/EditFormInstanceRecordMVCActionCommand.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.form.web.internal.portlet.action;
+
+import com.liferay.dynamic.data.mapping.constants.DDMPortletKeys;
+import com.liferay.dynamic.data.mapping.model.DDMFormInstance;
+import com.liferay.dynamic.data.mapping.service.DDMFormInstanceRecordLocalService;
+import com.liferay.dynamic.data.mapping.storage.DDMFormValues;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.util.ParamUtil;
+
+import javax.portlet.ActionRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Marcos Martins
+ */
+@Component(
+	immediate = true,
+	property = {
+		"javax.portlet.name=" + DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM_ADMIN,
+		"mvc.command.name=editFormInstanceRecord"
+	},
+	service = MVCActionCommand.class
+)
+public class EditFormInstanceRecordMVCActionCommand
+	extends AddFormInstanceRecordMVCActionCommand {
+
+	@Override
+	protected void saveFormInstanceRecord(
+			ActionRequest actionRequest, DDMFormInstance ddmFormInstance,
+			DDMFormValues ddmFormValues, long groupId,
+			ServiceContext serviceContext, long userId)
+		throws PortalException {
+
+		long ddmFormInstanceRecordId = ParamUtil.getLong(
+			actionRequest, "formInstanceRecordId");
+
+		ddmFormInstanceRecordLocalService.updateFormInstanceRecord(
+			userId, ddmFormInstanceRecordId, false, ddmFormValues,
+			serviceContext);
+	}
+
+	@Reference
+	protected DDMFormInstanceRecordLocalService
+		ddmFormInstanceRecordLocalService;
```

Comment:
@marcosapmf, we should use the `DDMFormInstanceRecordService` instead of the `DDMFormInstanceRecordLocalService` to guarantee that we'll check the user's permission.